src/components/ECheckbox.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import ELabel from "./ELabel.vue";
+type Props = {
+  modelValue?: boolean;
+  disabled?: boolean;
+  label?: string;
+};
+
+const { modelValue = false } = defineProps<Props>();
+
+const emit = defineEmits<{
+  (e: "update:modelValue", value: boolean): void; // add correct value type when you know it
+}>();
+
+const inputValue = computed({
+  get: () => modelValue,
+  set: (value) => {
+    emit("update:modelValue", value);
+  },
+});
+</script>
+
+<template>
+  <ELabel
+    v-if="label"
+    class="ECheckbox"
+    :label="label"
+    layout="horizontal"
+    :disabled="disabled"
```

Comment:
Kas siin ei peaks `:for="fieldId" olema.

fieldId voib tekitada ala sellise utiliidi abil

```
const randomString = (length) => [ ...Array(length) ].map(() => (~~(Math.random() * 36)).toString(36)).join('');
```

Reply:
imo for on ainult siis vajalik, kui input ei ole labeli sees. Seega siin ei ole vajalik.

Reply:
on sul mingi seisukoht, et kas label `for="fieldId"` või label inputi ümber?